src/poketch_data.c

```diff
@@ -0,0 +1,360 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "pokemon.h"
+#include "struct_decls/struct_021C0794_decl.h"
+
+#include "unk_020244AC.h"
+#include "poketch_data.h"
+#include "pokemon_icon.h"
+#include "savedata/save_table.h"
+
+#define POKETCH_REGISTRY_SIZE 32
+
+/**
+ * @brief All of the Poketch's internally tracked data including settings, registered apps, and the data for some apps (eg. pedometer, alarm clock).
+ */
+typedef struct PoketchData {
+    u8 unk_00_0 : 1;
+    u8 pedometerEnabled : 1;    //!< Whether or not the pedometer is registered and will take step count updates.
+    u8 unk_00_2 : 1;
+    u8 screenColor : 3;         //!< Screen palette color (valid values defined in poketch_data.h)
```

Comment:
In conjunction with the above comment, if the constants can be enums, this should also be able to be an enum type.